supabase/migrations/20240316051441_tables_add_threads_and_messages..sql

```diff
@@ -0,0 +1,39 @@
+CREATE TABLE IF NOT EXISTS threads (
+  id uuid NOT NULL PRIMARY KEY,
+  service_provider_id uuid NOT NULL,
+  customer_id uuid NOT NULL,
+  created_at timestamptz NOT NULL DEFAULT NOW(),
+  updated_at timestamptz NOT NULL DEFAULT NOW(),
+
+  FOREIGN KEY (service_provider_id) REFERENCES service_providers(user_id),
+  FOREIGN KEY (customer_id) REFERENCES customers(user_id)
+);
+
+ALTER TABLE threads
+  ENABLE ROW LEVEL SECURITY;
+
+CREATE POLICY "Can view own customer threads" ON threads
+  FOR SELECT USING (auth.uid() = customer_id);
+CREATE POLICY "Can update own service provider threads" ON threads
+  FOR UPDATE USING (auth.uid() = service_provider_id);
+
+CREATE TABLE IF NOT EXISTS messages (
+  id serial NOT NULL PRIMARY KEY,
+  thread_id uuid NOT NULL,
+  sender_id uuid NOT NULL,
+  recipient_id uuid NOT NULL,
+  content text NOT NULL,
+  sent_at timestamptz NOT NULL DEFAULT NOW(),
+  status text NOT NULL DEFAULT 'unread',
+
+  FOREIGN KEY (thread_id) REFERENCES threads(id),
+  FOREIGN KEY (sender_id) REFERENCES public.users(id)
+);
```

Comment:
In terms of how I want to access this data:
- `get all messages belonging to this thread`
- `get the latest message in this thread`, maybe, since we can broadcast the message via Supabase Realtime